File: code/python/ch18/practice/practice_05_jump_game_ii.py
"""
Practice 5: Jump Game II
==========================
Chapter 18: Greedy Algorithms — The Smart Shortcut

PROBLEM
-------
Given an array where each element is your max jump length,
find the minimum number of jumps to reach the last index.
(Guaranteed reachable.)

EXAMPLES
--------
>>> solve([2, 3, 1, 1, 4])
2
>>> solve([2, 3, 0, 1, 4])
2

CONSTRAINTS
-----------
- 1 <= len(nums) <= 10^4
- 0 <= nums[i] <= 1000
"""


def solve(nums: list[int]) -> int:
    """Return minimum number of jumps to reach the last index."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    nums = list(map(int, input().strip().split()))
    print(solve(nums))

File: code/cpp/ch18/practice/practice_05_jump_game_ii.cpp
/*
 * Practice 5: Jump Game II
 * Chapter 18: Greedy Algorithms — The Smart Shortcut
 *
 * PROBLEM: Minimum jumps to reach last index.
 * EXAMPLES: solve({2,3,1,1,4}) -> 2
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int solve(vector<int> nums) {
    // TODO: Replace this with your solution
    return 0;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n; cin >> n;
    vector<int> nums(n);
    for (int i = 0; i < n; i++) cin >> nums[i];
    cout << solve(nums) << endl;
    return 0;
}
